frontend/public/components/modals/add-users-modal.tsx

```diff
@@ -0,0 +1,48 @@
+import * as _ from 'lodash-es';
+import * as React from 'react';
+
+import { GroupModel } from '../../models';
+import { GroupKind, k8sPatch } from '../../module/k8s';
+import {
+  ModalBody,
+  ModalComponentProps,
+  ModalSubmitFooter,
+  ModalTitle,
+  createModalLauncher,
+} from '../factory/modal';
+import { ListInput, HandlePromiseProps, withHandlePromise } from '../utils';
+
+export const AddUsersModal = withHandlePromise((props: AddUsersModalProps) => {
```

Comment:
nit. What's our best practice for destructuring? I don't think it's consistent everywhere, but seems like in most cases we would destructure props here so that we don't have to use `props.[propName]` throughout the component.